File: noop-java/src/main/java/com/exyui/android/debugbottle/components/Injector/IntentInjector.java
package com.exyui.android.debugbottle.components.Injector;

import android.content.Intent;
import android.util.Log;

/**
 * Created by yuriel on 9/7/16.
 */

public class IntentInjector {

    public final static IntentInjector INSTANCE = new IntentInjector();
    private static final String TAG = "FakeIntentInjector";
    private IntentInjector() {}

    public void setActivity(Object a) {
        Log.d(TAG, "setActivity");
    }

    public void put(Object a, Object b) {
        Log.d(TAG, "put");
    }

    public Intent get() {
        return null;
    }

    public void run(Object a)  {
        Log.d(TAG, "run");
    }
}

File: noop-kotlin/src/main/kotlin/com/exyui/android/debugbottle/components/injector/IntentInjector.kt
package com.exyui.android.debugbottle.components.injector

import android.content.Intent
import android.util.Log

/**
 * Created by yuriel on 8/30/16.
 */
object IntentInjector {
    private val TAG = "FakeIntentInjector"

    fun setActivity(any: Any) {
        Log.d(TAG, "setActivity")
    }

    fun put(a: Any?, b: Any?) {
        Log.d(TAG, "add")
    }

    fun get(any: Any): Intent? = null

    fun run(any: Any) {
        Log.d(TAG, "run")
    }

}